server/netty-server/src/main/scala/sttp/tapir/server/netty/package.scala

```diff
@@ -1,19 +1,27 @@
 package sttp.tapir.server
 
 import io.netty.buffer.ByteBuf
+import io.netty.channel.{ChannelHandlerContext, ChannelPromise}
+import io.netty.handler.stream.{ChunkedFile, ChunkedStream}
 import sttp.monad.MonadError
 import sttp.monad.syntax._
 import sttp.tapir.server.model.ServerResponse
 
 import scala.concurrent.Future
 
 package object netty {
-  type Route[F[_]] = NettyServerRequest => F[Option[ServerResponse[ByteBuf]]]
+  type Route[F[_]] = NettyServerRequest => F[Option[ServerResponse[NettyResponse]]]
   type FutureRoute = Route[Future]
+  type NettyResponse = ChannelHandlerContext => NettyResponseContent
+
+  sealed abstract class NettyResponseContent(val channelPromise: ChannelPromise)
+  final case class ByteBufNettyResponseContent(override val channelPromise: ChannelPromise, byteBuf: ByteBuf) extends NettyResponseContent(channelPromise)
+  final case class ChunkedStreamNettyResponseContent(override val channelPromise: ChannelPromise, chunkedStream: ChunkedStream) extends NettyResponseContent(channelPromise)
+  final case class ChunkedFileNettyResponseContent(override val channelPromise: ChannelPromise, chunkedFile: ChunkedFile) extends NettyResponseContent(channelPromise)
```

Comment:
probably a cleaner encoding:

```
sealed trait NettyResponseContent {
  def channelPromise: ChannelPromise
}

object NettyResponseContent {
   case class AsByteBuf(channelPromise: ChannelPromise, byteBuf: ByteBuf) extends NettyResponseContent
  // ... etc
}
```

Reply:
Plus, since this is a prpoer type now, we should move it out of the package object to a top-level file

Reply:
updated